### PR TITLE
Upgrade Cadence to secure-cadence-m8

### PR DESCRIPTION
**Pull request by janezpodhostnik**
A simple bump in cadence version to [secure-cadence-m8](https://github.com/onflow/cadence/releases/tag/secure-cadence-m8)

### CODE REVIEW

**Review by SupunS — APPROVED** [review]

**Review by sideninja — APPROVED** [review]
LGTM

**Comment by janezpodhostnik** [comment]
bors merge

File: fvm/fvm_test.go [hunk 1]
```diff
@@ -525,7 +525,7 @@ func TestBlockContext_DeployContract(t *testing.T) {
 		err = vm.Run(ctx, tx, ledger, programs.NewEmptyPrograms())
 		require.NoError(t, err)
 
-		assert.NoError(t, tx.Err)
+		assert.ErrorContains(t, tx.Err, "program too ambiguous")
```
Comment (janezpodhostnik) [inline fvm/fvm_test.go:528]: This contract deployment now fails. I have changed the test to expect a "program too ambiguous" error. Is this ok/expected @turbolent.

Comment (janezpodhostnik) [reply fvm/fvm_test.go]: I used the solution from here https://axiomzen.slack.com/archives/CG0B7CJAJ/p1653502279521059

**Comment by janezpodhostnik** [comment]
bors merge

**Pull request merged**